src/part_file.cpp

```diff
@@ -208,18 +207,20 @@ namespace libtorrent
 		open_file(file::read_write | file::attribute_hidden, ec);
 		if (ec) return -1;
 
+		boost::shared_ptr<file> f = m_file;
 		l.unlock();
-		return m_file.readv(slot_offset(slot) + offset, bufs, num_bufs, ec);
+		return f->readv(slot_offset(slot) + offset, bufs, num_bufs, ec);
 	}
 
 	void part_file::open_file(int mode, error_code& ec)
 	{
-		if (m_file.is_open() && ((mode & file::rw_mask) == file::read_only
-			|| (m_file.open_mode() & file::rw_mask) == file::read_write))
+		if (m_file && m_file->is_open()
+			&& ((mode & file::rw_mask) == file::read_only
+			|| (m_file->open_mode() & file::rw_mask) == file::read_write))
 			return;
```

Comment:
Adding assertion for modes containing `lock_file` flag, could make sense, because it will block subsequent open operations if they request `read_write` access.